File: src/main/kotlin/info/hildegynoid/cue/converter/entity/CueFileInfo.kt
package info.hildegynoid.cue.converter.entity

import info.hildegynoid.cue.converter.loader.EOL
import info.hildegynoid.cue.converter.loader.INDENT
import info.hildegynoid.cue.converter.loader.escape

data class CueFileInfo(
    var fileName: String,
    var fileType: String,
    var tracks: MutableList<CueTrackInfo> = mutableListOf()
) {
    override fun toString(): String {
        return "FILE \"${fileName.escape()}\" $fileType" + EOL +
            tracks.joinToString("") { it.toString() }
                .prependIndent(INDENT)
    }
}

File: src/cue/entity/CueFileInfo.ts
import { EOL, escapeString, prependIndent } from '../Utils';
import { CueTrackInfo } from './CueTrackInfo';

export class CueFileInfo {
  fileName: string = '';

  fileType: string = '';

  tracks: Array<CueTrackInfo> = [];

  constructor(fileName: string, fileType: string) {
    this.fileName = fileName;
    this.fileType = fileType;
  }

  getTrackList(): Array<string> {
    const list = Array<string>();
    this.tracks.forEach((track) => {
      list.push(track.getTrack());
    });
    return list;
  }

  toString(): string {
    // return prependIndent(
    //   `FILE "${escapeString(this.fileName)}" ${
    //     this.fileType
    //   }${EOL}${this.tracks.join('')}`
    return `FILE "${escapeString(this.fileName)}" ${
      this.fileType
    }${EOL}${prependIndent(this.tracks.join(''))}`;
  }
}
